vault/token_store.go

```diff
@@ -2890,7 +2758,16 @@ func (ts *TokenStore) handleCreateCommon(ctx context.Context, req *logical.Reque
 		explicitEntityID = entity.ID
 	}
 
-	// Setup the token entry
+	// By using GetOk here, we set TokenEntry.Meta to nil instead of an empty map, when no meta was passed in the
+	// token creation API call. This maintains closer compatibility with previous behaviour of Vault, helping to avoid
+	// user surprise, and saving the need to update quite a few tests which expect this to be nil in such a case.
```

Comment:
This comment seems more useful for the PR reviewer rather than for the future readers/maintainers of the code. Consider removing it or at least shortening it to the first sentence (since `GetOk` seems to be a fairly established pattern throughout the code and the code below is quite self-explanatory).

Reply:
Well - the point I was trying to preserve for future readers, was why this particular parameter is using `GetOk` and a follow-up `if ok`, rather than just using the shorter `Get` to do it all on one line - like most of the other fields in this function do - and in fact, how I wrote the code originally.

Essentially, I'm recording why this code is not written in the natural way one would write it, if compatibility wasn't an issue.

`GetOk` is _usually_ for when code needs to get a field which might not be present in the request schema - but here it is being used solely to avoid coercing nil to an empty map.

Reply:
Comment rephrased to more directly express the core point.